x-pack/plugins/fleet/public/components/agent_enrollment_flyout/index.tsx

```diff
@@ -99,7 +99,9 @@ export const AgentEnrollmentFlyout: React.FunctionComponent<FlyOutProps> = ({
     }
   }, [selectedPolicy, isFleetServerPolicySelected]);
 
-  const { isK8s } = useIsK8sPolicy(selectedPolicy ? selectedPolicy : undefined);
+  const { isK8s, cloudSecurityIntegration } = useIsK8sPolicy(
+    selectedPolicy ? selectedPolicy : undefined
+  );
```

Comment:
Doesn't make much sense now to use the useIsK8sPolicy hook, let's split it into two hooks:


```js
export function useIsK8sPolicy(agentPolicy?: AgentPolicy) {
	const [isK8s, setIsK8s] = useState<K8sMode>('IS_LOADING');

  useEffect(() => {
    async function checkifK8s() {
      if (!agentPolicy) {
        setIsK8s('IS_LOADING');
        return;
      }

      setIsK8s(
        (agentPolicy.package_policies as PackagePolicy[]).some(isK8sPackage)
          ? 'IS_KUBERNETES'
          : 'IS_NOT_KUBERNETES'
      );
    }

    checkifK8s();
  }, [agentPolicy]);
	
	return { isK8s };
}
```

```js
export function useCloudSecurityIntegration(agentPolicy?: AgentPolicy) {
	const cloudSecurityIntegration = useMemo(() => {
		if (!agentPolicy) {
			return undefined;
		}

		const integrationType = getCloudSecurityIntegrationTypeFromPackagePolicy(agentPolicy);
		const cloudformationUrl = getCloudFormationTemplateUrlFromPackagePolicy(agentPolicy);

		return {
			integrationType,
			cloudformationUrl,
		};
	}, [agentPolicy]);

	return { cloudSecurityIntegration };
}
```

Then we can use like this:

```js
const { isK8s } = useIsK8sPolicy(selectedPolicy ?? undefined);
const { cloudSecurityIntegration } = useCloudSecurityIntegration(selectedPolicy ?? undefined);
```